shared/utility/math/euler.hpp

```diff
@@ -168,6 +168,92 @@ namespace utility::math::euler {
         return quat.matrix();
     }
 
+    /**
+     * @brief Convert rotation matrix to euler angles Thetaxy = [Roll, Pitch, Yaw]
+     */
+    template <typename Scalar>
+    void rot2rpy(const Eigen::Matrix<Scalar, Eigen::Dynamic, Eigen::Dynamic>& Rxy,
+                 Eigen::Matrix<Scalar, Eigen::Dynamic, 1>& Thetaxy) {
+        assert(Rxy.rows() == 3);
+        assert(Rxy.cols() == 3);
+        Thetaxy.resize(3, 1);
+        Thetaxy << atan2(Rxy(2, 1), Rxy(2, 2)), atan2(-Rxy(2, 0), sqrt(pow(Rxy(2, 1), 2) + pow(Rxy(2, 2), 2))),
+            atan2(Rxy(1, 0), Rxy(0, 0));
+    }
+
+    /**
+     * @brief Convert rotation matrix to euler angles Thetaxy = [Roll, Pitch, Yaw]
+     */
+    template <typename Scalar>
+    Eigen::Matrix<Scalar, Eigen::Dynamic, 1> rot2rpy(const Eigen::Matrix<Scalar, Eigen::Dynamic, Eigen::Dynamic>& Rxy) {
+        assert(Rxy.rows() == 3);
+        assert(Rxy.cols() == 3);
+        Eigen::Matrix<Scalar, Eigen::Dynamic, 1> Thetaxy;
+        Thetaxy.resize(3, 1);
+        Thetaxy << std::atan2(Rxy(2, 1), Rxy(2, 2)),
+            std::atan2(-Rxy(2, 0), std::sqrt(pow(Rxy(2, 1), 2) + std::pow(Rxy(2, 2), 2))),
+            std::atan2(Rxy(1, 0), Rxy(0, 0));
+        return Thetaxy;
+    }
+
+    /**
+     * @brief Rotation matrix around x-axis
+     */
+    template <typename Scalar>
+    void rotx(const Scalar& x, Eigen::Matrix<Scalar, Eigen::Dynamic, Eigen::Dynamic>& R) {
+        R.resize(3, 3);
+        R << 1, 0, 0, 0, cos(x), -sin(x), 0, sin(x), cos(x);
+    }
+
+    /**
+     * @brief Rotation matrix around y-axis
+     */
+    template <typename Scalar>
+    void roty(const Scalar& x, Eigen::Matrix<Scalar, Eigen::Dynamic, Eigen::Dynamic>& R) {
+        R.resize(3, 3);
+        R << cos(x), 0, sin(x), 0, 1, 0, -sin(x), 0, cos(x);
+    }
+
+    /**
+     * @brief Rotation matrix around z-axis
+     */
+    template <typename Scalar>
+    void rotz(const Scalar& x, Eigen::Matrix<Scalar, Eigen::Dynamic, Eigen::Dynamic>& R) {
+        R.resize(3, 3);
+        R << cos(x), -sin(x), 0, sin(x), cos(x), 0, 0, 0, 1;
+    }
+
+    /**
+     * @brief Convert euler angles Thetaxy = [Roll, Pitch, Yaw] to rotation matrix
+     */
+    template <typename Scalar>
+    void rpy2rot(const Eigen::Matrix<Scalar, Eigen::Dynamic, 1>& Thetaxy,
```

Comment:
Eigen has example code for making a rotation matrix from euler angles, that might be better to use?